packages/functions/src/providers/apy/BeefyAPYProvider.ts

```diff
@@ -0,0 +1,44 @@
+import { PluginData, Strategy } from '@midas-capital/types';
+import axios from 'axios';
+import { ExternalAPYProvider } from './ExternalAPYProvider';
+
+interface BeefyAPYResponse {
+  [key: string]: number;
+}
+
+class BeefyAPYProvider extends ExternalAPYProvider {
+  static apyEndpoint = 'https://api.beefy.finance/apy';
+  private beefyAPYs: BeefyAPYResponse | undefined;
+
+  constructor() {
+    super();
+  }
+
+  async getApy(pluginAddress: string, pluginData: PluginData): Promise<number> {
+    if (pluginData.strategy != Strategy.Beefy)
+      throw `BeefyAPYProvider: Not a Beefy Plugin ${pluginAddress}`;
+
+    if (!pluginData.apyDocsUrl) throw 'BeefyAPYProvider: `apyDocsUrl` is required to map Beefy APY';
+
+    const beefyID = pluginData.apyDocsUrl.split('/').pop();
+    if (!beefyID) throw 'BeefyAPYProvider: unable to extract `Beefy ID` from `apyDocsUrl`';
+
+    if (!this.beefyAPYs) {
+      await this._fetchBeefyAPYs();
+    }
+
+    const apy = this.beefyAPYs![beefyID];
+    if (!apy) throw `BeefyAPYProvider: Beefy ID: "${beefyID}" not included in Beefy APY data`;
+
+    return apy;
+  }
+
+  private async _fetchBeefyAPYs() {
```

Comment:
Probably would be a good idea to cache this call somehow -- though I believe we're already doing it with Supabase. 

Mind giving a sense of how often would this be called?

Reply:
It's just called once. As it's using the same instance of the BeefyAPYProvider. It will just call fetchBeefyAPY again if it's not available. Which should only be on the first request. If no BeefyPlugin is even available it will never be called.

Not very transparent for sure. Wasn't able to come up with a more transparent and enforced init code yet. Each APY Provider has to make sure on it's own it's to fetch as less as possible during multiple lookups.

https://github.com/Midas-Protocol/monorepo/pull/851/files#diff-6baebe0af0f07ff25c44719fa7224337f1b8b5d85847ba7a4545f6dc976ee624R26-R28